### PR TITLE
[IOTDB-1091] SDT improvement store last point

### PR DESCRIPTION
**Pull request by haimeiguo**
store last point when use sdt encoding
SDT compression is trigger by flushing, in current implementation the last point before flushing is not stored. 
The pr fixes this issue.

### CODE REVIEW

**Comment by jixuan1989** [comment]
Hi, could you add all PRs about the SDT to the SDT related jira ticket?

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java [hunk 1]
```diff
@@ -174,26 +174,26 @@ private void checkSdtEncoding() {
...
-      if (isSdtEncoding) {
-        //store last read time and value
-        time = sdtEncoder.getTime();
-        value = sdtEncoder.getLongValue();
-      }
+      pageWriter.write(isSdtEncoding ? sdtEncoder.getTime() : time,
+          isSdtEncoding ? sdtEncoder.getLongValue() : value);
+    }
+    // store last point for sdtEncoding
+    if (isSdtEncoding && isLastPoint) {
```
Comment (mychaow) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java:181]: last point may store twice?

**Review by mychaow — CHANGES_REQUESTED** [review]

**Review by mychaow — APPROVED** [review]

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java (hunk 1 above)
Comment (jixuan1989) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java:181]: if (! isSdtEncoding || isLastPoint) {
  pageWriter.write(time, value);
} else if (sdtEncoder.encodeLong(time, value)) {
 pageWriter.write(sdtEncoder.getTime(), sdtEncoder.getLongValue());
}

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java [hunk 2]
```diff
@@ -80,7 +80,7 @@
   private Statistics<?> statistics;
 
   private boolean isSdtEncoding;
-
+  private boolean isLastPoint;
```
Comment (jixuan1989) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java:83]: add comment: 
used for SDT. When the ChunkWriter WILL write the last data point in the chunk, set it to true to tell SDT saves the point.

**Review by jixuan1989 — CHANGES_REQUESTED** [review]
leave some commets to let others know how to use fields.

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java (hunk 1 above)
Comment (jixuan1989) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/write/chunk/ChunkWriterImpl.java:164]: any problem about my logic?

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**